std/src/interface/rgb25.rs

```diff
@@ -0,0 +1,214 @@
+// RGB standard library for working with smart contracts on Bitcoin & Lightning
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Written in 2023 by
+//     Dr Maxim Orlovsky <orlovsky@lnp-bp.org>
+//
+// Copyright (C) 2023 LNP/BP Standards Association. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused_braces)]
+
+use std::fmt::Debug;
+
+use bp::bc::stl::bitcoin_stl;
+use strict_encoding::{StrictDumb, StrictEncode};
+use strict_types::stl::std_stl;
+use strict_types::{CompileError, LibBuilder, TypeLib};
+
+use super::{
+    AssignIface, GenesisIface, GlobalIface, Iface, OwnedIface, Req, TransitionIface, VerNo,
+};
+use crate::interface::{ArgSpec, ContractIface};
+use crate::stl::{rgb_contract_stl, Amount, ContractData, Details, Name, Precision, StandardTypes};
+
+pub const LIB_NAME_RGB25: &str = "RGB25";
+/// Strict types id for the library providing data types for RGB25 interface.
+pub const LIB_ID_RGB25: &str = "sleep_shine_mono_2h6qQe4h857LhYbe5AS9ohU4iXNtPQfNzfeMDHEPsGBr";
+
+const SUPPLY_MISMATCH: u8 = 1;
+const NON_EQUAL_AMOUNTS: u8 = 2;
+const INVALID_PROOF: u8 = 3;
+const INSUFFICIENT_RESERVES: u8 = 4;
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+#[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
+#[strict_type(lib = LIB_NAME_RGB25, tags = repr, into_u8, try_from_u8)]
+#[repr(u8)]
+pub enum Error {
+    #[strict_type(dumb)]
+    SupplyMismatch = SUPPLY_MISMATCH,
+    NonEqualAmounts = NON_EQUAL_AMOUNTS,
+    InvalidProof = INVALID_PROOF,
+    InsufficientReserves = INSUFFICIENT_RESERVES,
+}
+
+fn _rgb25_stl() -> Result<TypeLib, CompileError> {
+    LibBuilder::new(libname!(LIB_NAME_RGB25), tiny_bset! {
+        std_stl().to_dependency(),
+        bitcoin_stl().to_dependency(),
+        rgb_contract_stl().to_dependency(),
+    })
+    .transpile::<Error>()
+    .compile()
+}
+
+/// Generates strict type library providing data types for RGB25 interface.
+pub fn rgb25_stl() -> TypeLib { _rgb25_stl().expect("invalid strict type RGB25 library") }
+
+pub fn rgb25() -> Iface {
+    let types = StandardTypes::with(rgb25_stl());
+
+    Iface {
+        version: VerNo::V1,
+        name: tn!("RGB25"),
+        global_state: tiny_bmap! {
+            fname!("name") => GlobalIface::required(types.get("RGBContract.Name")),
```

Comment:
Maybe we also need an optional ticker?

Reply:
By adding a ticker the interface would become very similar to RGB20, we prefer to keep this interface without ticker, indicating that RGB25 assets are not meant to represent "stocks", they are just collectible digital assets